platform/linuxbsd/os_linuxbsd.cpp

```diff
@@ -303,65 +303,150 @@ static String get_mountpoint(const String &p_path) {
-			trash_can = path;
+			trash_path = path;
 		}
 	}
 
-	// Otherwise, if ${XDG_DATA_HOME} is defined, use "${XDG_DATA_HOME}/Trash/files" as the trash can.
-	if (trash_can == "") {
+	// Otherwise, if ${XDG_DATA_HOME} is defined, use "${XDG_DATA_HOME}/Trash" as the trash can.
+	if (trash_path == "") {
 		char *dhome = getenv("XDG_DATA_HOME");
 		if (dhome) {
-			trash_can = String(dhome) + "/Trash/files";
+			trash_path = String(dhome) + "/Trash";
 		}
 	}
 
-	// Otherwise, if ${HOME} is defined, use "${HOME}/.local/share/Trash/files" as the trash can.
-	if (trash_can == "") {
+	// Otherwise, if ${HOME} is defined, use "${HOME}/.local/share/Trash" as the trash can.
+	if (trash_path == "") {
 		char *home = getenv("HOME");
 		if (home) {
-			trash_can = String(home) + "/.local/share/Trash/files";
+			trash_path = String(home) + "/.local/share/Trash";
 		}
 	}
 
 	// Issue an error if none of the previous locations is appropriate for the trash can.
-	if (trash_can == "") {
+	if (trash_path == "") {
 		ERR_PRINT("move_to_trash: Could not determine the trash can location");
 		return FAILED;
 	}
 
 	// Create needed directories for decided trash can location.
-	DirAccess *dir_access = DirAccess::create(DirAccess::ACCESS_FILESYSTEM);
-	Error err = dir_access->make_dir_recursive(trash_can);
-	memdelete(dir_access);
+	{
+		DirAccess *dir_access = DirAccess::create(DirAccess::ACCESS_FILESYSTEM);
+		Error err = dir_access->make_dir_recursive(trash_path);
+		Error err2 = dir_access->make_dir_recursive(trash_path + "/files");
+		Error err3 = dir_access->make_dir_recursive(trash_path + "/info");
+
+		// Issue an error if trash can is not created proprely.
+		if (err != OK || err2 != OK || err3 != OK) {
+			ERR_PRINT("move_to_trash: Could not create the trash can \"" + trash_path + "\"");
+			return err;
+		}
+		memdelete(dir_access);
+	}
 
-	// Issue an error if trash can is not created proprely.
-	if (err != OK) {
-		ERR_PRINT("move_to_trash: Could not create the trash can \"" + trash_can + "\"");
-		return err;
+	// The trash can is successfully created, now we check that we don't exceed our file name Length limit.
+	// If the file name it's too long trim it so we can add the identifyng number and ".trashinfo"
+	// Assumes that the file name Length limit is 255 characters.
+	String file_name = basename(p_path.utf8().get_data());
+	if (file_name.length() > 240) {
+		file_name = file_name.substr(0, file_name.length() - 15);
+	}
+
+	String dest_path = trash_path + "/files/" + file_name;
+	struct stat buff;
+	int id_number = 0;
+	String fn = file_name;
+
+	// Checks if a resource with the same name already exist in the trash can,
+	// if there is, add an identifying number to our resource's name.
+	while (stat(dest_path.utf8().get_data(), &buff) == 0) {
+		id_number++;
+
+		// Added a limit to check for identically named files already on the trash can
+		// if there are too many it could make the editor unresponsive.
+		if (id_number > 999) {
```

Comment:
You can use `ERR_FAIL_COND_V_MSG` to make this shorter.

```suggestion
		ERR_FAIL_COND_V_MSG(id_number > 999, FAILED, "Too many identically named resources already in the trash can.");
```

Reply:
Done.